### PR TITLE
Fix #22655: Port surface fixes from #19740 into parkpatch

### LINKED ISSUE #22655: Port #19740 surface element fixes to .parkpatch files
#19740 had several fixes for surface elements to be patched on scenarios that were held off until #21189 was merged. These can now be applied via the .parkpatch files.

### PR DESCRIPTION
**Pull request by tupaschoal**
Some things to consider:

- ~The "to_surface" is currently accepting integer indices, but if I recall correctly these are not constant, as they depend on the order of the objects being loaded. Can you confirm @Gymnasiast ? 
On my plugins I'm using `objectManager.load("rct2.terrain_surface.grid_yellow")` to make sure I get the proper index every time, but I'm not sure this is ideal for the patches since it would force load some surfaces and since the surfaces being used are already being loaded for the given scenario.~

- @HtotheTML I dropped the fixes for the tiles starting at X=39 and X=40, since they already seemed to be using the proper surface (sand). Note this PR makes the patches support the surface style changes, so it will be easier to go through alcatraz and other scenarios you said there are incorrect settings.

Before:
image

After: 
image

### CODE REVIEW

**Comment by Gymnasiast** [comment]
There is indeed no guarantee that object indices are always correct/the same. The best solution is let the parkpatch specify the JSON identifier, and let the C++ code find the object index.

**Comment by tupaschoal** [comment]
> There is indeed no guarantee that object indices are always correct/the same. The best solution is let the parkpatch specify the JSON identifier, and let the C++ code find the object index. It is possible to do so without forcibly loading them, although you obviously have to handle that correctly if it does happen.
> 
> If you can’t easily find how to do this, I am willing to look into it.

I managed to do it and it works properly, let me know if it was not how you were thinking about it.

File: src/openrct2/rct12/ScenarioPatcher.cpp [hunk 1]
```diff
@@ -56,6 +57,10 @@ static const std::string s_toKey = "to";
 static const std::string s_tilesKey = "tiles";
 static const std::string s_typeKey = "type";
 
+// Surface fix keys
+static const std::string s_surfacesKey = "surfaces";
+static const std::string s_destinationSurface = "to_surface";
```
Comment (Gymnasiast) [inline src/openrct2/rct12/ScenarioPatcher.cpp:62]: This does not adhere to our codestyle rules.
```suggestion
static const std::string _surfacesKey = "surfaces";
static const std::string _destinationSurface = "to_surface";
```

**Review by Gymnasiast — COMMENTED** [review]
Apart from one code style thing, this looks alright.

File: src/openrct2/rct12/ScenarioPatcher.cpp (hunk 1 above)
Comment (tupaschoal) [reply src/openrct2/rct12/ScenarioPatcher.cpp:62]: Thanks for the reminder, I've tackled them all on the second commit.

**Review by Gymnasiast — APPROVED** [review]
Code LGTM, but I haven’t tested it.

**Pull request merged**